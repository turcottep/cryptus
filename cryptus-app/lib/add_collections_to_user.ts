import UsernameSetting from "../components/basic/settings/username_setting/username_setting";
import findAllUsers from "../pages/api/users/findAllUsers";

export default async function add_collections_to_user(
  user_collections_addresses: string[],
  user_collections_slugs: string[],
  username
) {
  console.log("Adding Col");
  const res = await fetch("/api/users/addCollectionsAddress", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      addresses: user_collections_addresses,
      username: username,
    }),
  });
  const user = await res.json();
  return user;
}
