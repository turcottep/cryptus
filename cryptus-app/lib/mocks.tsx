import { nft, nft_collection } from "./data_types";

export const mock_collection: nft_collection = {
  id: 1,
  name: "CryptoKitties",
  description: "",
  image_url: "",
  external_url: "",
  market_cap: 0,
  address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
  nfts: [
    {
      name: "Savannah R1-R3, Sly R1",
      image_url:
        "https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8",
      description:
        "Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!",
      collection: "CryptoKitties",
      collection_size: null,
      collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
      token_id: "492809",
      external_url:
        "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/492809",
      last_sale_price: 0,
      last_sale_symbol: "ETH",
      properties: [
        { name: "mouth", value: "soserious", count: 214676, rarity: 0 },
        { name: "pattern", value: "spangled", count: 74400, rarity: 0 },
        {
          name: "virginity",
          value: "non-virgin",
          count: 980293,
          rarity: 0,
        },
        { name: "fur", value: "selkirk", count: 125552, rarity: 0 },
        { name: "eye_colour", value: "sapphire", count: 99039, rarity: 0 },
        { name: "accent_colour", value: "icy", count: 111490, rarity: 0 },
        {
          name: "highlight_colour",
          value: "coffee",
          count: 175517,
          rarity: 0,
        },
        {
          name: "cooldown_timer",
          value: "Catatonic (1 week)",
          count: 75674,
          rarity: 0,
        },
        { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
        {
          name: "base_colour",
          value: "mauveover",
          count: 146133,
          rarity: 0,
        },
        { name: "generation", value: 0, count: 36557, rarity: 0 },
        { name: "purrstige", value: "pu12", count: 19237, rarity: 0 },
        { name: "secret", value: "se6", count: 13121, rarity: 0 },
      ],
      rarity_rank: null,
    },
    {
      name: "Couple 69 2/2",
      image_url:
        "https://lh3.googleusercontent.com/lklxKijfeWuWTqU9ONM4U8T7vUXn2wuZYurL6kw-z2MfV82xzWVGzss0DzOTD_4QD2c0r3RQi8ibuK0egAN0Vzyw8w",
      description:
        "G'day. My name is Couple 69 2/2. My favourite colour is green. I like to tell stories, which are often either lawful or vivacious. Are you a magician? Because whenever I look at you, everyone else disappears!",
      collection: "CryptoKitties",
      collection_size: null,
      collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
      token_id: "739367",
      external_url:
        "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/739367",
      last_sale_price: 0,
      last_sale_symbol: "ETH",
      properties: [
        {
          name: "virginity",
          value: "non-virgin",
          count: 980293,
          rarity: 0,
        },
        { name: "mouth", value: "grim", count: 116805, rarity: 0 },
        { name: "pattern", value: "leopard", count: 65930, rarity: 0 },
        { name: "generation", value: 0, count: 36557, rarity: 0 },
        {
          name: "accent_colour",
          value: "sandalwood",
          count: 101193,
          rarity: 0,
        },
        {
          name: "base_colour",
          value: "greymatter",
          count: 189267,
          rarity: 0,
        },
        {
          name: "cooldown_timer",
          value: "Catatonic (1 week)",
          count: 75674,
          rarity: 0,
        },
        {
          name: "eye_colour",
          value: "thundergrey",
          count: 96833,
          rarity: 0,
        },
        { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
        {
          name: "highlight_colour",
          value: "egyptiankohl",
          count: 104341,
          rarity: 0,
        },
        { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
      ],
      rarity_rank: null,
    },
    {
      name: "Cyborg+Springcrocus; Caffeine",
      image_url:
        "https://lh3.googleusercontent.com/kPUtxNggeG8gMYaEjGMpRZACIGqR3IF3texGniVTI7lBxk8vz3m3LUM6bENWq1tlHjtKoYjqeGB1Y3yn-Im0sPBR7Sm7Rczxr0zYdw",
      description:
        "G'day. I am Cyborg+Springcrocus; Caffeine. I'm glad you've come. I can safely say that salsa is the greatest thing since sliced bread. Don't even try to convince me otherwise. I can't imagine being with any other breeder now that I've met you!",
      collection: "CryptoKitties",
      collection_size: null,
      collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
      token_id: "901479",
      external_url:
        "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/901479",
      last_sale_price: 0,
      last_sale_symbol: "ETH",
      properties: [
        { name: "mouth", value: "pouty", count: 287247, rarity: 0 },
        { name: "pattern", value: "rorschach", count: 90037, rarity: 0 },
        { name: "generation", value: 0, count: 36557, rarity: 0 },
        {
          name: "virginity",
          value: "non-virgin",
          count: 980293,
          rarity: 0,
        },
        {
          name: "highlight_colour",
          value: "coffee",
          count: 175517,
          rarity: 0,
        },
        {
          name: "accent_colour",
          value: "frosting",
          count: 86540,
          rarity: 0,
        },
        { name: "eye_shape", value: "wonky", count: 92220, rarity: 0 },
        { name: "base_colour", value: "brownies", count: 51182, rarity: 0 },
        {
          name: "cooldown_timer",
          value: "Swift (2 min)",
          count: 153623,
          rarity: 0,
        },
        { name: "eye_colour", value: "olive", count: 42178, rarity: 0 },
        { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
        { name: "purrstige", value: "pu8", count: 18824, rarity: 0 },
        { name: "secret", value: "se14", count: 6867, rarity: 0 },
      ],
      rarity_rank: null,
    },
  ],
};

export const mock_nft: nft = {
  name: "Savannah R1-R3, Sly R1",
  image_url:
    "https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8",
  description:
    "Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!",
  collection: "CryptoKitties",
  collection_size: null,
  collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
  token_id: "492809",
  external_url:
    "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/492809",
  last_sale_price: 0,
  last_sale_symbol: "ETH",
  properties: [
    { name: "mouth", value: "soserious", count: 214676, rarity: 0 },
    { name: "pattern", value: "spangled", count: 74400, rarity: 0 },
    {
      name: "virginity",
      value: "non-virgin",
      count: 980293,
      rarity: 0,
    },
    { name: "fur", value: "selkirk", count: 125552, rarity: 0 },
    { name: "eye_colour", value: "sapphire", count: 99039, rarity: 0 },
    { name: "accent_colour", value: "icy", count: 111490, rarity: 0 },
    {
      name: "highlight_colour",
      value: "coffee",
      count: 175517,
      rarity: 0,
    },
    {
      name: "cooldown_timer",
      value: "Catatonic (1 week)",
      count: 75674,
      rarity: 0,
    },
    { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
    {
      name: "base_colour",
      value: "mauveover",
      count: 146133,
      rarity: 0,
    },
    { name: "generation", value: 0, count: 36557, rarity: 0 },
    { name: "purrstige", value: "pu12", count: 19237, rarity: 0 },
    { name: "secret", value: "se6", count: 13121, rarity: 0 },
  ],
  rarity_rank: null,
};

export const mock_wallet: nft_collection[] = [
  {
    id: 1,
    name: "CryptoKitties",
    description: "",
    image_url: "",
    external_url: "",
    market_cap: 0,
    address: "1x06012c8cf97bead5deae237070f9587f8e7a266d",
    nfts: [
      {
        name: "Savannah R1-R3, Sly R1",
        image_url:
          "https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8",
        description:
          "Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "492809",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/492809",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "soserious", count: 214676, rarity: 0 },
          { name: "pattern", value: "spangled", count: 74400, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "fur", value: "selkirk", count: 125552, rarity: 0 },
          { name: "eye_colour", value: "sapphire", count: 99039, rarity: 0 },
          { name: "accent_colour", value: "icy", count: 111490, rarity: 0 },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "base_colour",
            value: "mauveover",
            count: 146133,
            rarity: 0,
          },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          { name: "purrstige", value: "pu12", count: 19237, rarity: 0 },
          { name: "secret", value: "se6", count: 13121, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Couple 69 2/2",
        image_url:
          "https://lh3.googleusercontent.com/lklxKijfeWuWTqU9ONM4U8T7vUXn2wuZYurL6kw-z2MfV82xzWVGzss0DzOTD_4QD2c0r3RQi8ibuK0egAN0Vzyw8w",
        description:
          "G'day. My name is Couple 69 2/2. My favourite colour is green. I like to tell stories, which are often either lawful or vivacious. Are you a magician? Because whenever I look at you, everyone else disappears!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "739367",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/739367",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "mouth", value: "grim", count: 116805, rarity: 0 },
          { name: "pattern", value: "leopard", count: 65930, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "accent_colour",
            value: "sandalwood",
            count: 101193,
            rarity: 0,
          },
          {
            name: "base_colour",
            value: "greymatter",
            count: 189267,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          {
            name: "eye_colour",
            value: "thundergrey",
            count: 96833,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "highlight_colour",
            value: "egyptiankohl",
            count: 104341,
            rarity: 0,
          },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Cyborg+Springcrocus; Caffeine",
        image_url:
          "https://lh3.googleusercontent.com/kPUtxNggeG8gMYaEjGMpRZACIGqR3IF3texGniVTI7lBxk8vz3m3LUM6bENWq1tlHjtKoYjqeGB1Y3yn-Im0sPBR7Sm7Rczxr0zYdw",
        description:
          "G'day. I am Cyborg+Springcrocus; Caffeine. I'm glad you've come. I can safely say that salsa is the greatest thing since sliced bread. Don't even try to convince me otherwise. I can't imagine being with any other breeder now that I've met you!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "901479",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/901479",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "pouty", count: 287247, rarity: 0 },
          { name: "pattern", value: "rorschach", count: 90037, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "accent_colour",
            value: "frosting",
            count: 86540,
            rarity: 0,
          },
          { name: "eye_shape", value: "wonky", count: 92220, rarity: 0 },
          { name: "base_colour", value: "brownies", count: 51182, rarity: 0 },
          {
            name: "cooldown_timer",
            value: "Swift (2 min)",
            count: 153623,
            rarity: 0,
          },
          { name: "eye_colour", value: "olive", count: 42178, rarity: 0 },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
          { name: "purrstige", value: "pu8", count: 18824, rarity: 0 },
          { name: "secret", value: "se14", count: 6867, rarity: 0 },
        ],
        rarity_rank: null,
      },
    ],
  },
  {
    id: 2,
    name: "CryptoKitties 2",
    description: "",
    image_url: "",
    external_url: "",
    market_cap: 0,
    address: "2x06012c8cf97bead5deae237070f9587f8e7a266d",
    nfts: [
      {
        name: "Savannah R1-R3, Sly R1",
        image_url:
          "https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8",
        description:
          "Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "492809",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/492809",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "soserious", count: 214676, rarity: 0 },
          { name: "pattern", value: "spangled", count: 74400, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "fur", value: "selkirk", count: 125552, rarity: 0 },
          { name: "eye_colour", value: "sapphire", count: 99039, rarity: 0 },
          { name: "accent_colour", value: "icy", count: 111490, rarity: 0 },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "base_colour",
            value: "mauveover",
            count: 146133,
            rarity: 0,
          },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          { name: "purrstige", value: "pu12", count: 19237, rarity: 0 },
          { name: "secret", value: "se6", count: 13121, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Couple 69 2/2",
        image_url:
          "https://lh3.googleusercontent.com/lklxKijfeWuWTqU9ONM4U8T7vUXn2wuZYurL6kw-z2MfV82xzWVGzss0DzOTD_4QD2c0r3RQi8ibuK0egAN0Vzyw8w",
        description:
          "G'day. My name is Couple 69 2/2. My favourite colour is green. I like to tell stories, which are often either lawful or vivacious. Are you a magician? Because whenever I look at you, everyone else disappears!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "739367",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/739367",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "mouth", value: "grim", count: 116805, rarity: 0 },
          { name: "pattern", value: "leopard", count: 65930, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "accent_colour",
            value: "sandalwood",
            count: 101193,
            rarity: 0,
          },
          {
            name: "base_colour",
            value: "greymatter",
            count: 189267,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          {
            name: "eye_colour",
            value: "thundergrey",
            count: 96833,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "highlight_colour",
            value: "egyptiankohl",
            count: 104341,
            rarity: 0,
          },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Cyborg+Springcrocus; Caffeine",
        image_url:
          "https://lh3.googleusercontent.com/kPUtxNggeG8gMYaEjGMpRZACIGqR3IF3texGniVTI7lBxk8vz3m3LUM6bENWq1tlHjtKoYjqeGB1Y3yn-Im0sPBR7Sm7Rczxr0zYdw",
        description:
          "G'day. I am Cyborg+Springcrocus; Caffeine. I'm glad you've come. I can safely say that salsa is the greatest thing since sliced bread. Don't even try to convince me otherwise. I can't imagine being with any other breeder now that I've met you!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "901479",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/901479",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "pouty", count: 287247, rarity: 0 },
          { name: "pattern", value: "rorschach", count: 90037, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "accent_colour",
            value: "frosting",
            count: 86540,
            rarity: 0,
          },
          { name: "eye_shape", value: "wonky", count: 92220, rarity: 0 },
          { name: "base_colour", value: "brownies", count: 51182, rarity: 0 },
          {
            name: "cooldown_timer",
            value: "Swift (2 min)",
            count: 153623,
            rarity: 0,
          },
          { name: "eye_colour", value: "olive", count: 42178, rarity: 0 },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
          { name: "purrstige", value: "pu8", count: 18824, rarity: 0 },
          { name: "secret", value: "se14", count: 6867, rarity: 0 },
        ],
        rarity_rank: null,
      },
    ],
  },
  {
    id: 3,
    name: "CryptoKitties 3",
    description: "",
    image_url: "",
    external_url: "",
    market_cap: 0,
    address: "3x06012c8cf97bead5deae237070f9587f8e7a266d",
    nfts: [
      {
        name: "Savannah R1-R3, Sly R1",
        image_url:
          "https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8",
        description:
          "Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "492809",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/492809",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "soserious", count: 214676, rarity: 0 },
          { name: "pattern", value: "spangled", count: 74400, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "fur", value: "selkirk", count: 125552, rarity: 0 },
          { name: "eye_colour", value: "sapphire", count: 99039, rarity: 0 },
          { name: "accent_colour", value: "icy", count: 111490, rarity: 0 },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "base_colour",
            value: "mauveover",
            count: 146133,
            rarity: 0,
          },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          { name: "purrstige", value: "pu12", count: 19237, rarity: 0 },
          { name: "secret", value: "se6", count: 13121, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Couple 69 2/2",
        image_url:
          "https://lh3.googleusercontent.com/lklxKijfeWuWTqU9ONM4U8T7vUXn2wuZYurL6kw-z2MfV82xzWVGzss0DzOTD_4QD2c0r3RQi8ibuK0egAN0Vzyw8w",
        description:
          "G'day. My name is Couple 69 2/2. My favourite colour is green. I like to tell stories, which are often either lawful or vivacious. Are you a magician? Because whenever I look at you, everyone else disappears!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "739367",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/739367",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          { name: "mouth", value: "grim", count: 116805, rarity: 0 },
          { name: "pattern", value: "leopard", count: 65930, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "accent_colour",
            value: "sandalwood",
            count: 101193,
            rarity: 0,
          },
          {
            name: "base_colour",
            value: "greymatter",
            count: 189267,
            rarity: 0,
          },
          {
            name: "cooldown_timer",
            value: "Catatonic (1 week)",
            count: 75674,
            rarity: 0,
          },
          {
            name: "eye_colour",
            value: "thundergrey",
            count: 96833,
            rarity: 0,
          },
          { name: "eye_shape", value: "chronic", count: 93145, rarity: 0 },
          {
            name: "highlight_colour",
            value: "egyptiankohl",
            count: 104341,
            rarity: 0,
          },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
        ],
        rarity_rank: null,
      },
      {
        name: "Cyborg+Springcrocus; Caffeine",
        image_url:
          "https://lh3.googleusercontent.com/kPUtxNggeG8gMYaEjGMpRZACIGqR3IF3texGniVTI7lBxk8vz3m3LUM6bENWq1tlHjtKoYjqeGB1Y3yn-Im0sPBR7Sm7Rczxr0zYdw",
        description:
          "G'day. I am Cyborg+Springcrocus; Caffeine. I'm glad you've come. I can safely say that salsa is the greatest thing since sliced bread. Don't even try to convince me otherwise. I can't imagine being with any other breeder now that I've met you!",
        collection: "CryptoKitties",
        collection_size: null,
        collection_address: "0x06012c8cf97bead5deae237070f9587f8e7a266d",
        token_id: "901479",
        external_url:
          "https://opensea.io/assets/0x06012c8cf97bead5deae237070f9587f8e7a266d/901479",
        last_sale_price: 0,
        last_sale_symbol: "ETH",
        properties: [
          { name: "mouth", value: "pouty", count: 287247, rarity: 0 },
          { name: "pattern", value: "rorschach", count: 90037, rarity: 0 },
          { name: "generation", value: 0, count: 36557, rarity: 0 },
          {
            name: "virginity",
            value: "non-virgin",
            count: 980293,
            rarity: 0,
          },
          {
            name: "highlight_colour",
            value: "coffee",
            count: 175517,
            rarity: 0,
          },
          {
            name: "accent_colour",
            value: "frosting",
            count: 86540,
            rarity: 0,
          },
          { name: "eye_shape", value: "wonky", count: 92220, rarity: 0 },
          { name: "base_colour", value: "brownies", count: 51182, rarity: 0 },
          {
            name: "cooldown_timer",
            value: "Swift (2 min)",
            count: 153623,
            rarity: 0,
          },
          { name: "eye_colour", value: "olive", count: 42178, rarity: 0 },
          { name: "fur", value: "ragdoll", count: 173014, rarity: 0 },
          { name: "purrstige", value: "pu8", count: 18824, rarity: 0 },
          { name: "secret", value: "se14", count: 6867, rarity: 0 },
        ],
        rarity_rank: null,
      },
    ],
  },
];

import React, { useState, useEffect } from "react";

type State = {
  email: "";
  pinnumber: "";
  password: "";
  confirmpassword: "";
};

export function getMockProps(StepID: number) {
  const [step, setStep] = useState<number>(StepID);
  const [state, setState] = useState<State>({
    email: "",
    pinnumber: "",
    password: "",
    confirmpassword: "",
  });

  // Proceed to next step
  const nextStep = () => {
    setStep(step + 1);
  };

  // Go back to prev step
  const prevStep = () => {
    setStep(step - 1);
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    const target_id = e.target.id;
    setState({ ...state, [target_id]: newValue });
  };

  return {
    nextStep: nextStep,
    prevStep: prevStep,
    handleChange: handleChange,
    values: state,
    step: step,
  };
}

// DO NOT DELETE PLEASE THANK YOU
/* import { signIn } from "next-auth/react";
import FindUserIdFromWalletAdress from "../lib/findUserIdFromWalletAdress";
import CreateAccountFromWalletAddress from "../lib/createAccountFromWalletAddress";
import FindUserFromUserId from "../lib/findUserFromUserId";
declare var window: any;

export function getMockCallback(LoadState) {
  const [loading, setLoading] = useState<Boolean>(LoadState);

  const connectMetamask = async () => {
    // router.push("login?");
    setLoading(true);
    if (!window.ethereum) {
      console.log("please donwload MetaMask");
      window.open("https://metamask.io/", "_blank").focus();
      setLoading(false);
    } else {
      try {
        await window.ethereum.enable();
        const accounts = await window.ethereum.request({
          method: "eth_requestAccounts",
        });
        const wallet_address = accounts[0];

        const userId = await FindUserIdFromWalletAdress(wallet_address, false);

        if (!userId) {
          //Create Account with this wallet address
          const user = await CreateAccountFromWalletAddress(
            wallet_address,
            false
          );
          signIn("credentials", {
            redirect: true,
            address: wallet_address,
            callbackUrl: `${window.location.origin}/edit_profile`,
          });
        } else {
          const user = await FindUserFromUserId(userId, false, false);
          signIn("credentials", {
            redirect: true,
            address: wallet_address,
            callbackUrl: `${window.location.origin}/` + user.username,
          });
        }
      } catch (error) {
        // console.error(error);
        setLoading(false);
        // router.push("login?error=CancelMetamask");
      }
    }
  };

  return {
    cb: connectMetamask,
  };
} */
