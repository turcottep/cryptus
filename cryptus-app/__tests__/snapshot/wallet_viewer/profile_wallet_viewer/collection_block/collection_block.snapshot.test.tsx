import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

import { mock_collection } from "../../../../../lib/mocks";
import CollectionBlock from "../../../../../components/wallet_viewer/profile_wallet_viewer/collection_block/collection_block";

describe("<CollectionBlock />", () => {
  it("should go to collection_details", async () => {
    /* const handleLink = jest.fn();
        const { findByTestId } = render(<CollectionBlock {...nft_collection}/>); 
  
        const collection = await findByTestId("collection_block");

        fireEvent.click(collection);

        expect(handleLink).toBeCalledWith({ */
  });

  it("should render correctly", () => {
    const tree = renderer
      .create(<CollectionBlock nft={mock_collection} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
