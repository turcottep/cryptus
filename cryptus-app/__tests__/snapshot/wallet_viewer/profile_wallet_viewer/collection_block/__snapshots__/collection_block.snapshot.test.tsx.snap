// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionBlock /> should render correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <div>
          <div>
            <div>
              <img
                alt="Listen up! Savannah R1-R3, Sly R1 here. I'm here to enjoy licking doorknobs and kickboxing. When my owner isn't watching, I steal their mom jeans and hoard them in my litter box. I'm not sorry. Let's pawty!"
                src="https://lh3.googleusercontent.com/j2tpXbSVHCzhtM2betJkxxlug7Y2i7yC9fyPPiJCTIBqxpU4VxsGk7Osw04lt8TetpSOp2uamKE_Ppy1fC_2cr8"
              />
            </div>
          </div>
          <div>
            <div>
              <img
                alt="G'day. My name is Couple 69 2/2. My favourite colour is green. I like to tell stories, which are often either lawful or vivacious. Are you a magician? Because whenever I look at you, everyone else disappears!"
                src="https://lh3.googleusercontent.com/lklxKijfeWuWTqU9ONM4U8T7vUXn2wuZYurL6kw-z2MfV82xzWVGzss0DzOTD_4QD2c0r3RQi8ibuK0egAN0Vzyw8w"
              />
            </div>
          </div>
          <div>
            <div>
              <img
                alt="G'day. I am Cyborg+Springcrocus; Caffeine. I'm glad you've come. I can safely say that salsa is the greatest thing since sliced bread. Don't even try to convince me otherwise. I can't imagine being with any other breeder now that I've met you!"
                src="https://lh3.googleusercontent.com/kPUtxNggeG8gMYaEjGMpRZACIGqR3IF3texGniVTI7lBxk8vz3m3LUM6bENWq1tlHjtKoYjqeGB1Y3yn-Im0sPBR7Sm7Rczxr0zYdw"
              />
            </div>
          </div>
          <div>
            <div>
              <div>
                <div>
                  <img
                    alt=""
                    src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                  />
                </div>
              </div>
              <div>
                <div>
                  <img
                    alt=""
                    src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                  />
                </div>
              </div>
              <div>
                <div>
                  <img
                    alt=""
                    src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                  />
                </div>
              </div>
              <div>
                <div>
                  <img
                    alt=""
                    src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      CryptoKitties
    </div>
  </div>
</div>
`;
